tekton/pipeline_run_test.go

```diff
@@ -132,5 +164,55 @@ var _ = Describe("PipelineRun", func() {
 				To(Equal(reflect.TypeOf(&tektonv1beta1.PipelineRun{})))
 		})
 
+		It("Can add ApplicationSnapshot as json to the PipelineRun", func() {
+			Expect(k8sClient.Create(ctx, snapshot)).Should(Succeed())
+
+			snapshot.TypeMeta.APIVersion = apiVersion
+			snapshot.TypeMeta.Kind = "ApplicationSnapshot"
+
+			releasePipelineRun.WithApplicationSnapshot(snapshot)
+
+			Expect(releasePipelineRun.Spec.Params[0].Name).To(Equal("applicationSnapshot"))
+			Expect(json.Unmarshal(
+				[]byte(releasePipelineRun.Spec.Params[0].Value.StringVal),
+				&unmarshaledApplicationSnapshotSpec)).Should(Succeed())
+
+			// check if the unmarshaled data has what we expect
+			Expect(unmarshaledApplicationSnapshotSpec.Application).To(Equal("testapplication"))
+		})
+
+		It("Can add the ReleaseStrategy to the PipelineRun", func() {
+			strategy.Spec.Pipeline = releasePipelineRun.AsPipelineRun().Name
```

Comment:
You could set it to any random value like `release-pipeline` when you declare it and skip this line.

Reply:
The actual use case it to use the name from the strategy so let's stick to it. The tests it's ok though. It's just the value used.